src/Features/Core/Portable/ExtractInterface/AbstractExtractInterfaceService.cs

```diff
@@ -270,20 +353,30 @@ public ExtractInterfaceResult ExtractInterfaceFromAnalyzedType(ExtractInterfaceT
             return unformattedInterfaceDocument;
         }
 
-        private static Solution GetSolutionWithFormattedInterfaceDocument(Document unformattedInterfaceDocument, CancellationToken cancellationToken)
+        private Solution GetFormattedSolution(Solution unformattedSolution, IEnumerable<DocumentId> documentIds, CancellationToken cancellationToken)
         {
-            Solution solutionWithInterfaceDocument;
-            var formattedRoot = Formatter.Format(unformattedInterfaceDocument.GetSyntaxRootSynchronously(cancellationToken),
-                unformattedInterfaceDocument.Project.Solution.Workspace, cancellationToken: cancellationToken);
-            var rootToSimplify = formattedRoot.WithAdditionalAnnotations(Simplifier.Annotation);
-            var finalInterfaceDocument = Simplifier.ReduceAsync(unformattedInterfaceDocument.WithSyntaxRoot(rootToSimplify), cancellationToken: cancellationToken).WaitAndGetResult_CanCallOnBackground(cancellationToken);
-
-            solutionWithInterfaceDocument = finalInterfaceDocument.Project.Solution;
-            return solutionWithInterfaceDocument;
+            // Since code action performs formatting and simplification on a single document, 
+            // this ensures that anything marked with formatter or simplifier annotations gets 
+            // correctly handled as long as it it's in the listed documents
+            var formattedSolution = unformattedSolution;
+            foreach (var documentId in documentIds)
+            {
+                var document = formattedSolution.GetDocument(documentId);
+                var formattedRoot = Formatter.Format(document.GetSyntaxRootSynchronously(cancellationToken),
+                    Formatter.Annotation,
+                    document.Project.Solution.Workspace, 
+                    cancellationToken: cancellationToken);
+
+                var formattedDocument = document.WithSyntaxRoot(formattedRoot);
+                var simplifiedDocument = Simplifier.ReduceAsync(formattedDocument, Simplifier.Annotation, cancellationToken: cancellationToken).WaitAndGetResult_CanCallOnBackground(cancellationToken);
```

Comment:
is there reason why all these manipulation is using WaitAndGetResult and using GetSyntaxRootSynchronously?

Reply:
When I tried to propogate Async up I was hitting threading issues with the tests. Async is the right thing to do here, but the code was already synchronous. I'm continuing the trend because I wasn't able to reasonably get async working and it's not a requirement to add this feature.

Reply:
@ryzngard You should be able to make almost all of the impl code async.  Where you'd need the explicit `Wait` would be the very top level call at the UI layer.  This is a pattern we use elsewhere without problem.  I can help you with this if you need more guidance here.

Reply:
@heejaechang and I plan to look over this today and see where I went wrong. My point was that it shouldn't block this checkin, since the code before had similar behavior. I'll do what I can to do the right thing though.